cmd/internal/serverutil/main.go

```diff
@@ -226,13 +226,13 @@ func AnnounceSelf(ctx context.Context, client *clientv3.Client, etcdService, end
 	if err != nil {
 		glog.Exitf("Failed to create etcd manager: %v", err)
 	}
-	fullEndpoint := fmt.Sprintf("%s/%s", etcdService, endpoint)
-	em.AddEndpoint(ctx, fullEndpoint, endpoints.Endpoint{Addr: endpoint})
+	fullEndpoint := fmt.Sprintf("%s/%s", etcdService, endpointName)
```

Comment:
It looks like if you put `endpointAddr` instead of `endpointName` here, it will work as previously.

Reply:
I'm judging by this guide: https://etcd.io/docs/v3.4.0/dev-guide/grpc_naming/

Writing `("my-service", naming.Update{Op: naming.Add, Addr: "1.2.3.4", Metadata: "..."})` resulted in a `"my-service/1.2.3.4"` key.

Reply:
But the new guide has a misleading "/e1" suffix which I'm not sure how to interpret: https://etcd.io/docs/current/dev-guide/grpc_naming/

Reply:
This documentation is what led me down this adventure, thinking I was being really smart in making the key unique. I'm now less sure, and think that this is only useful for master-election type of stuff. For service discovery, having a common prefix may actually be more useful. i.e. this PR should just be reverted. I still need to step back and consider this when the world gives me an hour without a high priority interrupt :-)